src/request.ts

```diff
@@ -339,20 +339,14 @@ class DatastoreRequest {
    *   const apiResponse = data[0];
    * });
    */
-  delete(keys: Entities):
-      void|Promise<google.datastore.v1.Datastore.CommitCallback>;
-  delete(
-      keys: Entities,
-      callback: google.datastore.v1.Datastore.CommitCallback): void;
-  delete(
-      keys: Entities, gaxOptions: CallOptions,
-      callback: google.datastore.v1.Datastore.CommitCallback): void;
+  delete(): Promise<google.datastore.v1.CommitResponse>;
```

Comment:
Still don't think this type is right.  Probably should be `Promise<CommitResponse>`